openedx_learning/core/itemstore/models_api.py

```diff
@@ -0,0 +1,21 @@
+from django.db import models
+
+from .models import ItemVersion
+
+
+class ItemVersionDataMixin(models.Model):
+    """
+    Minimal abstract model to let people attach data to ItemVersions.
+
+    The idea is that if you have a model that is associated with a specific
+    version of an item, the join is going to be 1:1 with an ItemVersion, and
+    potentially M:1 with your data model.
```

Comment:
> is Data immutable

It should be or Bad Things will happen.

> is ItemVersionData:Data an M:1 relationship because multiple versions of an Item may reference the same underlying Data?

Yes. One of the underlying assumptions here is that if you have some data that's associated with an ItemVersion, changing that data means that the Item has changed in some way, and thus forms a new ItemVersion. So if you change the retry-settings or some such for an Item, there's a new ItemVersion. But for all the other apps that didn't change their data, we can just make new entries that join the new ItemVersion with the same data that those apps had for the last ItemVersion.

Reply:
@ormsbee 

> It should be or Bad Things will happen.

Cool, I assumed so but wanted to be sure. Is there a way to indicate that, like marking it read-only in Django admin?

> Yes. ...

Makes sense 👍🏻 It would be good to include that explanation somewhere in the library before releasing v1.0. Not a blocker for this PR; I know you're still iterating.

Reply:
> Cool, I assumed so but wanted to be sure. Is there a way to indicate that, like marking it read-only in Django admin?

Yeah, I'll definitely do that when I start making some admin pages.